File: Strategy Summit/Number Theory/Q1. Find Greatest Common Divisor of Array/Q1_FindGreatestCommonDivisorofArray.java
class Solution {
    public int findGCD(int[] nums) {

        // Sort the array to easily get minimum and maximum elements
        Arrays.sort(nums);

        // 'a' stores the maximum element
        int a = nums[nums.length - 1];

        // 'b' stores the minimum element
        int b = nums[0];

        // Apply Euclidean Algorithm to find GCD
        while (b > 0) {

            // Store current value of b
            int temp = b;

            // Update b to remainder of a divided by b
            b = a % b;

            // Update a to previous b
            a = temp;
        }

        // 'a' now contains the GCD of min and max element
        return a;
    }
}

File: Strategy Summit/Number Theory/Q1. Find Greatest Common Divisor of Array/Q1_FindGreatestCommonDivisorofArray.py
class Solution(object):
    def findGCD(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """

        # Get the minimum and maximum elements from the array
        a, b = min(nums), max(nums)

        # Use subtraction-based Euclidean Algorithm
        # Keep reducing until both numbers become equal
        while a != b:

            # Subtract smaller number from larger number
            b = b - a

            # Reassign a as smaller and b as larger again
            a, b = min(a, b), max(a, b)

        # When both are equal, that value is the GCD
        return a